docs/core/advanced_tutorials/using-result-handlers.md

```diff
@@ -0,0 +1,140 @@
+
+Let's take a look at using result handlers. As introduced in the concept documentation ["Results and Result Handlers"](../concepts/results.md), Prefect tasks associate their return value with a `Result` object attached to a task's `State`, and they can be extended with `ResultHandler`s that, when enabled, persist results to storage as a pickle into one of the supported storage backends.
+
+Why would you use result handlers in the first place? One common situation is to allow for inspection of intermediary data when debugging flows (including -- maybe especially -- production ones). By checkpointing data during different data transformation Tasks, runtime data bugs can be tracked down by analyzing the checkpoints sequentially.
+
+## Setting up to handle results
+
+Checkpointing must be enabled for result handlers to be called. You must enable in two places: globally for the Prefect installation, and at the task level by providing a result handler to tasks either through its flow initialization or as a task-level override. Depending on which results handlers you configure, you may also need to ensure your authentication credentials are set up properly. For Cloud users on Prefect 0.9.1+, some of this is handled by default.
+
+For Core-only users or Cloud users on Prefect versions <0.9.1, you must:
+
+- Opt-in to checkpointing globally by setting the `prefect.config.flows.checkpointing` to "True"
+- Specify the result handler your tasks will use for at least one level of specificity (flow-level or task-level)
+
+For Cloud users on Prefect version 0.9.1+:
+
+- Checkpointing will automatically be turned on; you can disable it by passing `checkpoint=False` to each task you want to turn it off for
+- A result handler that matches the storage backend of your `prefect.config.flows.storage` setting will automatically be applied to all tasks, if available; notably this is not yet supported for Docker Storage
+- You can override the automatic result handler at the global level, flow level, or task level
+
+#### Setting result handler at the flow level
+```bash
+export PREFECT__FLOWS__CHECKPOINTING=true
+```
+
+```python{8-9}
+# flow.py
+from prefect.engine.result_handlers import LocalResultHandler
+
+@task
+def add(x, y=1):
+   return x + y
+
+# send the configuration to the Flow object
+with Flow("my handled flow!", result_handler=LocalResultHandler()):
+   first_result = add(1, y=2)
+   second_result = add(x=first_result, y=100)
+```
+
+#### Setting result handler at the task level
+
+```bash
+export PREFECT__FLOWS__CHECKPOINTING=true
+```
+
+```python{4-5,11-12}
+# flow.py
+from prefect.engine.result_handlers import LocalResultHandler
+
+# configure on the task decorator
+@task(result_handler=LocalResultHandler())
+def add(x, y=1):
+   return x + y
+
+with Flow("my handled flow!"):
+   first_result = add(1, y=2)
+   # or send as a keyword argument at task initialization
+   second_result = add(x=first_result, y=100, result_handler=LocalResultHandler())
```

Comment:
The `@task` decorator initializes the task under the hood, so passing result handler here will raise an error

Reply:
Resolved in https://github.com/PrefectHQ/prefect/pull/2076/commits/aefb44725e5438057742fa60040e605502c135af. Changed it to refer to the class instantiation kwargs as the other option for tasks which is possibly what I was trying to go for originally